### ISSUE TITLE
Livetime info is misleading?

### ISSUE THREAD

**Issue by luca-giunti** [gammapy/gammapy#3903]
**Gammapy version**
0.19

**Bug description**
I find that the current handling of the `livetime` entry in the Map/SpectrumDataset `info_dict` is a bit misleading. It can lead to unexpected situations like the following: an empty dataset cutout or a spectrum dataset extracted from an empty region can have nonzero `livetime`, since the latter is propagated by simply adding up the `livetime` value every time that two datasets are stacked. I show this in an [example notebook](https://github.com/luca-giunti/share/blob/master/debug.ipynb). 

Basically, the `livetime` value shown in the `info_dict` corresponds to the total livetime of the parent stacked dataset, which is well defined but (I think) a bit useless. In any case it should perhaps not be propagated as-is to cutouts or spectrum datasets.

**Comment by adonath** [comment]
Thank @luca-giunti! I agree storing the livetime as a single number is not the best solution as it gives rise to issue like the one you reported here. I remember we discussed a similar issue in the context of stacking datasets before. @registerrier proposed to not stack the livetime if the mask safe was all zero. However this is not a good solution either, because the different value of even a single pixel would change the behavior. I think the only correct and reasonable solution is to work with a livetime map here. It‘s the only way to store the information correctly and pass it on to other datasets.

**Comment by registerrier** [comment]
Livetime maps creation functions have been introduced. We can close now.
